apps/console/src/features/roles/utils/role-management-utils.ts

```diff
@@ -87,32 +72,25 @@ export class RoleManagementUtils {
                     : response.data;
 
                 let permissionTree: TreeNode[] = [];
-                let isStartingWithTwoNodes: boolean = false;
 
-                permissionTree = permissionStringArray.reduce((arr, path, index) => {
+                permissionTree = permissionStringArray.reduce((arr, path) => {
 
                     let nodes: TreeNode[] = [];
 
-                    if(index === 0 && path.resourcePath.replace(/^\/|\/$/g, "").split("/").length == 2) {
-                        isStartingWithTwoNodes = true;
-                    }
-                    if (isStartingWithTwoNodes) {
-                        nodes = generatePermissionTree(
-                            path, 
-                            RoleManagementUtils.permissionJoining(path.resourcePath.replace(/^\/|\/$/g, ""), 2),
-                            arr
-                        );
-                    } else {
-                        nodes = generatePermissionTree(
-                            path, 
-                            path.resourcePath.replace(/^\/|\/$/g, "").split("/"),
-                            arr
-                        );
-                    }
+                    nodes = generatePermissionTree(
+                        path,
+                        path.resourcePath.replace(/^\/|\/$/g, "").split("/"),
+                        arr
+                    );
                     
                     return nodes;
                 },[]);
 
+                if (tenantDomain !== "carbon.super"
```

Comment:
This could impact sub-organizations. Is that an expected outcome?

Reply:
Also, let's get the super tenant from the constants.

```
AppConstants.getSuperTenant()
```

Reply:
We are using https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/organizations/utils/organization-role-management-utils.ts for organization permissions so this file shouldn't impact sub organizations @thivi. @kavinda1995 please confirm.

Reply:
@brionmario Fixed with https://github.com/wso2/identity-apps/pull/3304/commits/c5d0a6acc34ad6d7fd075650d99c5f01927635a5